src/main_app/components/SignUp/index.js

```diff
@@ -1,65 +1,100 @@
-import { Formik, Field, Form } from 'formik'
+import { useState } from 'react'
+import { loader } from 'graphql.macro'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
 import * as Yup from 'yup'
+import { useMutation, Mutation } from '@apollo/client'
+
+import client from 'services/graphql/client'
 
 const sleep = (ms) => new Promise((r) => setTimeout(r, ms))
 
+const exampleMutation = loader('../../graphql/createUser.graphql')
+
 const SignupSchema = Yup.object().shape({
-    firstName: Yup.string()
-      .min(4, 'Too Short!')
-      .max(50, 'Too Long!')
-      .required('Required'),
-    lastName: Yup.string()
-      .min(2, 'Too Short!')
-      .max(50, 'Too Long!')
-      .required('Required'),
-    email: Yup.string().email('Invalid email').required('Required')
+  firstName: Yup.string()
+    .min(4, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  lastName: Yup.string()
+    .min(2, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  email: Yup.string().email('Invalid email').required('Required'),
+  password: Yup.string()
+    .min(8, 'Too Short!')
+    .max(12, 'Too Long!')
+    .required('Required')
 })
 
-const SignUp = () => (
-    <div>
-        <h1>Sign Up</h1>
-        <Formik
-        initialValues={{
-            firstName: '',
-            lastName: '',
-            email: '',
-            password: '',
-        }}
-        validationSchema={SignupSchema}
-        onSubmit={async (values, actions) => {
-            await sleep(500);
-            alert(JSON.stringify(values, null, 2));
-            actions.setSubmitting(false);
-        }}
-        >
-        {({ isSubmitting, errors, touched }) => (
-            <Form>
+const SignUp = () => {
+  const [state, setState] = useState ({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: ""
+  })
```

Comment:
I see you're using this state to hold the values of the form. Don't do this! Formik will store the values internally and you'll receive them on the `onSubmit` event.